src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -497,6 +514,90 @@ public function getArrayType(Expr\Array_ $expr, callable $getTypeCallback): Type
 		return $arrayType;
 	}
 
+	/**
+	 * @param callable(Expr): Type $getTypeCallback
+	 *
+	 * @throws OversizedConstantArrayTypeException
+	 */
+	private function builtDegradedConstantArray(Array_ $node, callable $getTypeCallback): Type
+	{
+		$isList = true;
+		$isEmptyArray = true;
+		$isKeysNonEmptyString = true;
+		$isValuesNonEmptyString = true;
+
+		$valueTypes = [];
+		$itemKeyTypes = [];
+		foreach ($node->items as $item) {
+			if ($item === null) {
+				continue;
+			}
+
+			$isEmptyArray = false;
+
+			$key = $item->key;
+			if ($key !== null) {
+				$isList = false;
+
+				$itemKeyType = $getTypeCallback($key);
+				if (!$itemKeyType instanceof ConstantScalarType || $key instanceof ClassConstFetch) {
```

Comment:
We skip const class fetch, as I was not able to make ArrayType accept a OversizedArrayType with reduced precision when using Constant enumerations like `self::SOME_*`.
See failing test when removing this check